cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -377,6 +453,33 @@ impl YaccParser {
                 i = self.parse_ws(j, true)?;
                 continue;
             }
+            if let Some(j) = self.lookahead_is("%expect-unused", i) {
+                i = self.parse_ws(j, false)?;
+                while i < self.src.len() {
+                    if self.lookahead_is("%", i).is_some() {
```

Comment:
I *think* this will allow `%expect-unused %` to be parsed successfully, which doesn't feel quite right.

Reply:
Indeed.

Reply:
I added a test for this, the way it works is that since `%` is just lookahead, it tries to parse it as a declaration on the next iteration and then throws an `UnknownDeclaration`.